src/main/kotlin/com/exactpro/th2/codec/Main.kt

```diff
@@ -56,46 +58,37 @@ class CodecCommand : CliktCommand() {
 
     override fun run() = runBlocking {
         try {
-            val commonFactory =  (if (configs == null)
-                                                    CommonFactory()
-                                                else
-                                                    CommonFactory.createFromArguments("--configs=" + configs))
-                .also {
-                    resources.add {
-                        logger.info("Closing common factory")
-                        it.close()
-                    }
+            val commonFactory = when (configs) {
+                null -> CommonFactory()
+                else -> CommonFactory.createFromArguments("--configs=$configs")
+            }.apply {
+                resources.add {
+                    logger.info("Closing common factory")
+                    close()
                 }
+            }
 
-            val configuration = Configuration.create(commonFactory, sailfishCodecConfig)
+            val configuration = Configuration.create(commonFactory, codecSettings ?: error("Path to codec settings file is not specified"))
```

Comment:
Codec settings might be `null` if I am not mistaken. Some codecs don't need any additional parameters